packages/common/rollup-configs.js

```diff
@@ -1,20 +1,56 @@
+import { execSync } from 'child_process';
 import commonjs from '@rollup/plugin-commonjs';
 import nodeResolve from '@rollup/plugin-node-resolve';
 import typescript from '@rollup/plugin-typescript';
 import analyzer from 'rollup-plugin-analyzer';
 import dts from 'rollup-plugin-dts';
 import css from 'rollup-plugin-import-css';
-import { removeFiles } from './rollup-plugins/removeFiles';
 import { replaceCode } from './rollup-plugins/replaceCode';
-import { writeBuildMetadata } from './rollup-plugins/writeBuildMetadata';
-
-/** @typedef {import("type-fest").PackageJson} PackageJson */
+import { writeJSONFile } from './rollup-plugins/writeJSONFile';
 
 // https://yarnpkg.com/advanced/lifecycle-scripts#environment-variables
 const rootDir = process.env.PROJECT_CWD;
 
 /**
- * @param {PackageJson} pkg
+ * @param {import('type-fest').PackageJson} pkg
+ */
+const getBuildMetadata = (pkg) => {
+  const now = new Date();
+
+  return {
+    packageName: pkg.name,
+    packageVersion: pkg.version,
+    buildDate: now.toLocaleString('en-US', { dateStyle: 'long' }),
+    buildTime: now.toLocaleString('en-US', { timeStyle: 'long' }),
+    gitCommit: execSync('git rev-parse HEAD').toString().trim(),
+    gitBranch: execSync('git rev-parse --abbrev-ref HEAD').toString().trim(),
+  };
+};
+
+/**
+ * @param {import('type-fest').PackageJson} pkg
+ * @param {Record<string, string>} buildMetadata
+ */
+const getBanner = (pkg, buildMetadata) => {
+  const padLength = Object.keys(buildMetadata).reduce(
+    (maxLength, key) => (key.length > maxLength ? key.length : maxLength),
+    0,
+  );
+
+  const text = `
+  OpenShift Dynamic Plugin SDK
+  ${pkg.repository.url.replace(/\.git$/, '')}
+
+  ${Object.entries(buildMetadata)
+    .map(([key, value]) => `${key.padEnd(padLength)} : ${value}`)
```

Comment:
This allows for better human-readable output:

![](https://user-images.githubusercontent.com/648971/186248040-44391179-4a68-4b25-903a-ea8615811902.png)